src/services/contact-notifier.js

```diff
@@ -12,7 +16,88 @@ module.exports = {
     return contacts.map(contact => `${contact.givenName} ${contact.familyName} <${contact.email}>`);
   },
 
-  async send({ to, subject, html }) {
+  /**
+   * Schedules (or updates) a campaign notification
+   */
+  async schedule({
+    type,
+    campaignId,
+    to,
+    cc,
+    subject,
+    html,
+    sendAt,
+  }) {
+    const criteria = { type, campaignId, status: { $in: ['Pending', 'Sent'] } };
+    const notification = await CampaignNotification.findOne(criteria);
+    if (notification) {
+      notification.set({ to, cc, sendAt });
+      return notification.save();
+    }
+    return CampaignNotification.create({
+      type,
+      campaignId,
+      to,
+      cc,
+      subject,
+      html,
+      sendAt,
+    });
+  },
+
+  init() {
+    // Run every 10 minutes
+    const job = new CronJob({
```

Comment:
There should be some protection here to prevent multiple calls to `init`, there-by preventing the same job from running multiple times